doc/user/refutation.md

```diff
@@ -0,0 +1,86 @@
+Refutation of Facts
+===================
+
+This document describes how Icicle resovles fact conflicts, e.g. from duplciate facts.
+
+Each fact may be defined with a refutation expression, which acts as a `nubBy` expression on the facts stream.
+
+Examples
+--------
+
+### No refutation
+
+This is the default. Facts are consumed as-is. Duplicate ingestion will result in duplicate facts.
+
+
+### Time
+
+```
+[fact.keyed_by_time]
+  key = "time"
+  encoding="int"
+  
+bulbasaur|keyed_by_time|1|2014-01-01
+bulbasaur|keyed_by_time|3|2014-01-01
+bulbasaur|keyed_by_time|5|2016-01-01
+charmander|keyed_by_time|0|2014-01-01
+charmander|keyed_by_time|2|2014-01-01
+```
+
+```
+> feature keyed_by_time ~> count value
+[bulbasaur, 2,charmander, 1]
+
+> feature keyed_by_time ~> newest value
+[bulbasaur, 5,charmander, 0]
+```
+
+### Expression: `year_of time`
+
+```
+[fact.keyed_by_year_of_time]
+  key = "year_of time"
+  encoding="string"
+
+blatoise|keyed_by_year_of_time|foo14|2014-01-01
+blatoise|keyed_by_year_of_time|foo15|2015-01-01
+blatoise|keyed_by_year_of_time|foo16|2016-01-01
+blatoise|keyed_by_year_of_time|foo160302|2016-03-02
+blatoise|keyed_by_year_of_time|foo160307|2016-03-07
+```
+
+```
+> feature keyed_by_year_of_time ~> count value
+[blatoise, 3]
+
+> feature keyed_by_year_of_time ~> newest value
+[blatoise, "foo16"]
+
+> feature keyed_by_year_of_time ~> group value ~> newest time
+[blatoise, [("foo14",2014-01-01)
+           ,("foo15",2015-01-01)
+           ,("foo16",2016-01-01)]]
+```
+
+### Struct Values
+
+```
+[fact.capitalism]
+  key = "transaction_id"
+  encoding="(transaction_id:int,dollarydoos:int)"
+
+zubat|capitalism|{"transaction_id":0,"dollarydoos":2}|2015-01-01
+zubat|capitalism|{"transaction_id":3,"dollarydoos":1}|2016-01-01
+zubat|capitalism|{"transaction_id":3,"dollarydoos":9}|2016-01-05
+```
+
+```
+> feature capitalism ~> count dollarydoos
+[zubat, 2]
+
+> feature capitalism ~> newest dollarydoos
+[zubat, 1]
```

Comment:
The idea was that the highest priority facts will come first, this makes it easier to implement sparse states because you don't have to undo a field that you may have already set, if that makes sense?